src/components/Navbar.js

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useRef } from 'react';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import { useKanban } from '../hooks/useKanban';
+
+export function Navbar() {
+  const { user, signInWithGoogle, signOut } = useAuth();
+  const { isDark, toggleTheme } = useTheme();
+  const { boardTitle, updateBoardTitle, addTask, updateColumn, columns } = useKanban();
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(boardTitle);
+  const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [showForms, setShowForms] = useState(false);
+  const titleInputRef = useRef(null);
+
+  const handleTitleClick = () => {
+    if (user) {
+      setIsEditingTitle(true);
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleChange = (e) => {
+const handleTitleBlur = async () => {
+const debouncedSetLoading = debounce((value) => setIsLoading(value), 300);
+debouncedSetLoading(true);
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
useEffect(() => {
  const debouncedSetLoading = debounce((value) => setIsLoading(value), 300);
  return () => {
    debouncedSetLoading.cancel();
  };
}, []);
```
The debounced function should be cleaned up when the component unmounts to prevent memory leaks and potential state updates on unmounted components.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;